File: api/Models/Store.cs
using Microsoft.EntityFrameworkCore;
using Microsoft.EntityFrameworkCore.Metadata.Internal;
using System.ComponentModel.DataAnnotations;

namespace api.Models
{
    [Index(nameof(Ruc), IsUnique = true)]
    public class Store
    {
        public int Id { get; set; }
        [Required]
        [MaxLength(100)]
        public string Name { get; set; } = default!;
        [Required]
        [MaxLength(20)]
        public string Ruc { get; set; } = default!;
        [Required]
        [MaxLength(150)]
        public string Address { get; set; } = default!;
        [Required]
        [MaxLength(15)]
        public string Phone { get; set; } = default!;
        public ICollection<ApplicationUser> Users { get; set; } = new List<ApplicationUser>();
        public ICollection<StoreProduct> StoreProducts { get; set; } = new List<StoreProduct>();
        public ICollection<BuyOrder> BuyOrders { get; set; } = new List<BuyOrder>();
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
        public DateTime? DeletedAt { get; set; }
    }
}

File: api/app/models/store.ts
import { DateTime } from 'luxon';
import { BaseModel, column, hasMany } from '@adonisjs/lucid/orm';
import BuyOrder from './buy_order.js';
import StoreProduct from './store_product.js';
import Sale from './sale.js';
import type { HasMany } from '@adonisjs/lucid/types/relations';
import User from './user.js';

export default class Store extends BaseModel {
  @column({ isPrimary: true })
  declare id: number;

  @column()
  declare name: string;

  @column()
  declare ruc: string;

  @column()
  declare address: string;

  @column()
  declare phone: string;

  @column.dateTime({ autoCreate: true })
  declare createdAt: DateTime;

  @column.dateTime({ autoCreate: true, autoUpdate: true })
  declare updatedAt: DateTime;

  @column.dateTime()
  declare deletedAt: DateTime | null;

  @hasMany(() => BuyOrder)
  declare buyOrders: HasMany<typeof BuyOrder>;

  @hasMany(() => StoreProduct)
  declare storeProducts: HasMany<typeof StoreProduct>;

  @hasMany(() => Sale)
  declare sales: HasMany<typeof Sale>;

  @hasMany(() => User)
  declare users: HasMany<typeof User>;
}
